autoemulate/printing.py

```diff
@@ -34,3 +45,84 @@ def _print_cv_results(models, scores_df, model=None, sort_by="r2"):
         scores = get_model_scores(scores_df, model)
         print(f"Scores for {model} across all folds:")
         print(scores)
+
+
+def _print_setup(cls):
+    """Print the setup of the AutoEmulate object.
+
+    If in an IPython session, the setup will be displayed as an HTML table.
+
+    Parameters
+    ----------
+    cls : AutoEmulate
+        The AutoEmulate object.
+    """
+    if not cls.is_set_up:
+        raise RuntimeError("Must run setup() before print_setup()")
+
+    models = "\n- " + "\n- ".join(
+        [
+            x[1].__class__.__name__
+            for pipeline in cls.models
+            for x in pipeline.steps
+            if x[0] == "model"
+        ]
+    )
+    metrics = "\n- " + "\n- ".join([metric.__name__ for metric in cls.metrics])
+
+    settings = pd.DataFrame(
+        [
+            str(cls.X.shape),
+            str(cls.y.shape),
+            str(cls.train_idxs.shape[0]),
+            str(cls.test_idxs.shape[0]),
+            str(cls.param_search),
+            str(cls.search_type),
+            str(cls.param_search_iters),
+            str(cls.scale),
+            str(cls.scaler.__class__.__name__ if cls.scaler is not None else "None"),
+            str(cls.reduce_dim),
+            str(
+                cls.dim_reducer.__class__.__name__
+                if cls.dim_reducer is not None
+                else "None"
+            ),
+            str(cls.cv.__class__.__name__ if cls.cv is not None else "None"),
+            str(cls.folds),
+            str(cls.n_jobs if cls.n_jobs is not None else "1"),
+        ],
+        index=[
+            "Simulation input shape (X)",
+            "Simulation output shape (y)",
+            "# training set samples (train_idxs)",
+            "# test set samples (test_idxs)",
+            "Do hyperparameter search (param_search)",
+            "Type of hyperparameter search (search_type)",
+            "# sampled parameter settings (param_search_iters)",
+            "Scale data before fitting (scale)",
+            "Scaler (scaler)",
+            "Dimensionality reduction before fitting (reduce_dim)",
+            "Dimensionality reduction method (dim_reducer)",
+            "Cross-validation strategy (cv)",
+            "# folds (folds)",
+            "# parallel jobs (n_jobs)",
```

Comment:
one after thought here: Maybe this would be most useful if the argument names (rather than attribute names) would be in brackets. Would you agree?

If so, the two things to change are:

```
"# training set samples (train_idxs)",
"# test set samples (test_idxs)",
```
to 
` "test set size (test_set_size)" `
(and the value could be 20% or 0.2)

and 
`"Cross-validation strategy (cv)",`
to
`"Cross-validation strategy (fold_strategy)",`

Reply:
Ah, that makes sense. I was thinking that these should refer to any properties that exist on the object (hence, `cv` instead of `fold_strategy` as `self.cv` has been set; same with `train_idxs` and `test_idxs` as those are also existing as properties, whereas `test_set_size` is not a property on the object)... But since we're just showing the setup settings, I suppose it makes sense!

Reply:
See: https://github.com/alan-turing-institute/autoemulate/pull/195/commits/c0504defae06b2b9b7ca633a7f6a0b6824dfa3e2

Reply:
ok, thanks @kallewesterling ! Feel free to merge!